### ISSUE TITLE
Installing 1.7.1 under Windows and jruby 9.1.5.0 requires OCI8 and fails

### ISSUE THREAD

**Issue by pandar1** [rsim/oracle-enhanced#992]
Despite the following statement in the change log of 1.7.0.BETA1
"add_dependency with ruby-oci8 only if it runs cruby, not jruby [#902]"
I'm experiencing a similar issue in the current version, under windows, trying to install via a gem install

```
C:\jrails4_update\applications\assetsolutions>gem install activerecord-oracle_enhanced-adapter
Building native extensions.  This could take a while...
C:/jrails4_update/jruby-9.1.5.0/lib/ruby/stdlib/rubygems/ext/ext_conf_builder.rb
:56: warning: Tempfile#unlink or delete called on open file; ignoring
ERROR:  Error installing activerecord-oracle_enhanced-adapter:
        ERROR: Failed to build gem native extension.

current directory: C:/jrails4_update/jruby-9.1.5.0/lib/ruby/gems/shared/gems/ruby-oci8-2.2.2/ext/oci8
C:/jrails4_update/jruby-9.1.5.0/bin/jruby.exe -r ./siteconf20160912-11172-1tzjrig.rb extconf.rb
checking for load library path...
  LD_LIBRARY_PATH is not set.
checking for cc... ---------------------------------------------------
Error Message:
  The compiler failed to generate an executable file.
  You have to install development tools first.

Backtrace:
  C:/jrails4_update/jruby-9.1.5.0/lib/ruby/stdlib/mkmf.rb:456:in `try_do'
  C:/jrails4_update/jruby-9.1.5.0/lib/ruby/stdlib/mkmf.rb:541:in `try_link0'
  C:/jrails4_update/jruby-9.1.5.0/lib/ruby/stdlib/mkmf.rb:864:in `try_run'
  C:/jrails4_update/jruby-9.1.5.0/lib/ruby/gems/shared/gems/ruby-oci8-2.2.2/ext/oci8/oraconf.rb:635:in `check_cc'
  C:/jrails4_update/jruby-9.1.5.0/lib/ruby/gems/shared/gems/ruby-oci8-2.2.2/ext/oci8/oraconf.rb:625:in `init'
  C:/jrails4_update/jruby-9.1.5.0/lib/ruby/gems/shared/gems/ruby-oci8-2.2.2/ext/oci8/oraconf.rb:777:in `initialize'
  C:/jrails4_update/jruby-9.1.5.0/lib/ruby/gems/shared/gems/ruby-oci8-2.2.2/ext/oci8/oraconf.rb:320:in `get'  extconf.rb:22:in `<main>'
---------------------------------------------------
See:
 * http://www.rubydoc.info/github/kubo/ruby-oci8/file/docs/install-full-client.md for Oracle full client
 * http://www.rubydoc.info/github/kubo/ruby-oci8/file/docs/install-instant-client.md for Oracle instant client
 * http://www.rubydoc.info/github/kubo/ruby-oci8/file/docs/install-on-osx.md for OS X
 * http://www.rubydoc.info/github/kubo/ruby-oci8/file/docs/report-installation-issue.md to report an issue.

RuntimeError: RuntimeError
     get at C:/jrails4_update/jruby-9.1.5.0/lib/ruby/gems/shared/gems/ruby-oci8-2.2.2/ext/oci8/oraconf.rb:345
  <main> at extconf.rb:22
*** extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of necessary
libraries and/or headers.  Check the mkmf.log file for more details.  You may
need configuration options.
```

**Comment by pandar1** [comment]
I managed to install it by downloading the gem, editing the metadata file and removing the dependencies to oci8, installing from local and editing the gemfile, removing the ifs and enabling the code for jruby

**Comment by pandar1** [comment]
Will this issue be fixed in the next update? it seems that the check misses to recognize jruby in the latest version... is this a jruby problem not setting the environment variable under windows or a problem of the check? thanks

**Comment by yahonda** [comment]
Thanks for reporting.

Looks like rubygems does not support platform specific dependencies https://github.com/rubygems/rubygems/issues/1492 . It suggests to create separate gemspec files for each platform but I do not take approach.

Instead I'm going to remove this entry `s.add_dependency(%q<ruby-oci8>, [">= 2.2.0"]) if !defined?(RUBY_ENGINE) || RUBY_ENGINE == 'ruby'` to address it. Also updating read me to add `ruby-oci8` explicitly in each Gemfile.

**Comment by yahonda** [comment]
Oracle enhanced adapter 1.7.2 has been released including #995. 

I have also confirmed that Oracle enhanced adapter can be installed with JRuby 9.1.5 on Windows 7

```
C:\Windows\System32>jruby -v
jruby 9.1.5.0 (2.3.1) 2016-09-07 036ce39 Java HotSpot(TM) 64-Bit Server VM 25.71
-b15 on 1.8.0_71-b15 +jit [mswin32-x86_64]

C:\Windows\System32>gem install activerecord-oracle_enhanced-adapter
Fetching: activerecord-oracle_enhanced-adapter-1.7.2.gem (100%)
Successfully installed activerecord-oracle_enhanced-adapter-1.7.2
1 gem installed
```

Closing #992. If you are still getting the same error, please reopen it.
Thanks again for using Oracle enhanced adapter and reporting this issue.
